indexmanager/registry.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021 The Rode Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package indexmanager
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"strings"
+)
+
+//counterfeiter:generate -o ../mocks . MappingsRegistry
+type MappingsRegistry interface {
+	// LoadMappings reads the index mapping and version from JSON file in the directory specified in Config.MappingsPath.
+	LoadMappings() error
+	// IndexName returns the full index name, using the prefix, version, inner name, and document kind.
+	IndexName(documentKind, inner string) string
```

Comment:
`inner`/`innerName` is the best I could come up with for the part of the index name that's not the prefix, version, or document kind.